lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -2295,6 +2297,298 @@ spec = describe "SHARED_TRANSACTIONS" $ do
+            verify rGet
+                [ expectField
+                        (#balance . #total) (`shouldBe` Quantity transfer)
+                , expectField
+                        (#balance . #available) (`shouldBe` Quantity transfer)
+                ]
+
+        -- child shared wallets delegate to different pools and get rewards
+        -- after several epochs
+        pool1:pool2:_ <- map (view $ _Unwrapped . #id) . snd <$>
+            unsafeRequest @[ApiT StakePool]
+            ctx (Link.listStakePools arbitraryStake) Empty
+
+        let delegationJoin pool = Json [json|{
+                "delegations": [{
+                    "join": {
+                        "pool": #{ApiT pool},
+                        "stake_key_index": "0H"
+                    }
+                }]
+            }|]
+        -- one child shared wallet delegating to pool1
+        rTx3 <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shared walActive1) Default
+            (delegationJoin pool1)
+        verify rTx3
+            [ expectResponseCode HTTP.status202 ]
+        let (ApiSerialisedTransaction apiTx3 _) =
+                getFromResponse #transaction rTx3
+        signedTx3 <-
+            signSharedTx ctx walActive1 apiTx3
+                [ expectResponseCode HTTP.status202 ]
+        submittedTx3 <- submitSharedTxWithWid ctx walActive1 signedTx3
+        verify submittedTx3
+            [ expectResponseCode HTTP.status202
+            ]
+
+        -- one child shared wallet delegating to pool2
+        rTx4 <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shared walActive2) Default
+            (delegationJoin pool2)
+        verify rTx4
+            [ expectResponseCode HTTP.status202 ]
+        let (ApiSerialisedTransaction apiTx4 _) =
+                getFromResponse #transaction rTx4
+        signedTx4 <-
+            signSharedTx ctx walActive2 apiTx4
+                [ expectResponseCode HTTP.status202 ]
+        submittedTx4 <- submitSharedTxWithWid ctx walActive2 signedTx4
+        verify submittedTx4
+            [ expectResponseCode HTTP.status202
+            ]
+
+        -- checking awards
+        eventually "Shared Wallet 1 is delegating to pool1" $ do
+            request @ApiWallet ctx (Link.getWallet @'Shared walActive1) Default Empty
+                >>= flip verify
+                    [ expectField #delegation (`shouldBe` delegating (ApiT pool1) [])
+                    ]
+        eventually "Shared Wallet 2 is delegating to pool2" $ do
+            request @ApiWallet ctx (Link.getWallet @'Shared walActive2) Default Empty
+                >>= flip verify
+                    [ expectField #delegation (`shouldBe` delegating (ApiT pool2) [])
+                    ]
+
+        waitForNextEpoch ctx
+        waitForNextEpoch ctx
+        waitForNextEpoch ctx
+        waitForNextEpoch ctx
+        waitForNextEpoch ctx
+        waitForNextEpoch ctx
```

Comment:
Do we need that many epochs? :thinking:

Reply:
just wanted to make sure taken rewards compensate several tx fees every run